extensions/cornerstone/src/components/WindowLevelActionMenu/WindowLevelActionMenu.tsx

```diff
@@ -70,60 +198,85 @@ export function WindowLevelActionMenu({
-                    color: '#fff',
-                    maxNumTicks: 8,
-                    tickSize: 5,
-                    tickWidth: 1,
-                    labelMargin: 3,
+              onSetColorbar({
+                viewportId,
+                options: {
+                  colormaps,
+                  ticks: {
+                    position: colorbarTickPosition,
                   },
+                  width: colorbarWidth,
+                  position: colorbarContainerPosition,
+                  activeColormapName: colorbarInitialColormap,
                 },
               });
             }}
           />
         </div>
+        {colormaps && (
+          <AllInOneMenu.SubMenu
+            key="colorLUTPresets"
+            itemLabel="Color LUT"
+            itemIcon="icon-color-lut"
+          >
+            <div className="all-in-one-menu-item flex w-full justify-center">
+              <ButtonGroup
+                buttons={buttons}
+                onActiveIndexChange={index => {}}
+                defaultActiveIndex={0}
+                className="w-[70%]"
+              />
+            </div>
+            <div className="all-in-one-menu-item flex w-full justify-center">
+              <SwitchButton
+                label="Preview in viewport"
+                checked={showPreview}
+                onChange={checked => {
+                  setShowPreview(checked);
+                }}
+              />
+            </div>
+            <AllInOneMenu.DividerItem />
+            <AllInOneMenu.ItemPanel>
+              {colormaps.map((colormap, index) => (
+                <AllInOneMenu.Item
+                  key={index}
+                  label={colormap.description}
+                  onClick={() => {
+                    onSetColorLUT({ viewportId, colormap });
+                    setPrePreviewColormap(getCurrentViewportColormap(viewportId));
+                  }}
+                  onMouseEnter={() => {
+                    if (showPreviewRef.current) {
+                      setPrePreviewColormap(getCurrentViewportColormap(viewportId));
+                      onSetColorLUT({ viewportId, colormap });
+                    }
+                  }}
+                  onMouseLeave={() => {
+                    if (showPreviewRef.current && prePreviewColormapRef.current) {
+                      onSetColorLUT({ viewportId, colormap: prePreviewColormapRef.current });
+                    }
+                  }}
+                ></AllInOneMenu.Item>
+              ))}
+            </AllInOneMenu.ItemPanel>
+          </AllInOneMenu.SubMenu>
+        )}
         {presets && (
           <AllInOneMenu.SubMenu
             key="windowLevelPresets"
             itemLabel={t('Modality Window Presets', { modality: Object.keys(presets)[0] })}
+            itemIcon="viewport-window-level"
             headerComponent={
               <AllInOneMenu.HeaderItem>
                 {t('Modality Presets', { modality: Object.keys(presets)[0] })}
```

Comment:
we need to refactor this, this is getting out of hand, can you please split into couple of components? I guess the useEffect logics for colorbar does not need to touch window level preset component and vice versa